backend/layers/processing/process_validate_atac.py

```diff
@@ -0,0 +1,219 @@
+            self.logger.info(f"Updated database with {artifact_type}.")
+            self.update_processing_status(dataset_version_id, processing_status_key, DatasetConversionStatus.UPLOADED)
+            return artifact_id
+        except Exception as e:
+            self.logger.error(e)
+            raise ConversionFailed(processing_status_key) from None
+
+    def skip_atac_validation(
+        self, local_anndata_filename: str, manifest: IngestionManifest, dataset_version_id
+    ) -> bool:
+        """
+        Check if atac validation should be skipped
+        :param local_anndata_filename: the local anndata file
+        :param manifest: the manifest
+        :param dataset_version_id: the dataset version id
+        :return: True if the validation should be skipped, False otherwise
+        """
+        # check if the anndata should have a fragment file
+        try:
+            result = self.schema_validator.check_anndata_requires_fragment(local_anndata_filename)
+        except ValueError as e:  # fragment file forbidden
+            self.logger.warning(f"Anndata does not support atac fragment files for the follow reason: {e}")
+            if manifest.atac_fragment:
+                self.update_processing_status(
+                    dataset_version_id,
+                    DatasetStatusKey.VALIDATION,
+                    DatasetValidationStatus.INVALID,
+                )
+                raise ValidationAtacFailed(errors=[str(e), "Fragment file not allowed for non atac anndata."]) from None
+            self.logger.warning("Skipping fragment validation")
+            self.update_processing_status(
+                dataset_version_id,
+                DatasetStatusKey.ATAC_FRAGMENT,
+                DatasetConversionStatus.SKIPPED,
+                validation_errors=[str(e)],
+            )
+            return True
+
+        if manifest.atac_fragment is None:
+            if result:  # fragment file required
+                self.update_processing_status(
+                    dataset_version_id, DatasetStatusKey.VALIDATION, DatasetValidationStatus.INVALID
+                )
+                raise ValidationAtacFailed(errors=["Anndata requires fragment file"])
+            else:  # fragment file optional
+                self.logger.info("Fragment is optional and not present. Skipping fragment validation.")
+                self.update_processing_status(
+                    dataset_version_id,
+                    DatasetStatusKey.ATAC_FRAGMENT,
+                    DatasetConversionStatus.SKIPPED,
+                    validation_errors=["Fragment is optional and not present."],
+                )
+                return True
+        return False
+
+    def hash_file(self, file_name: str) -> str:
+        """
+        Hash the file
+        :param file_name: the file to hash
+        :return: the hash
+        """
+        with open(file_name, "rb") as f:
+            return hashlib.sha256(f.read()).hexdigest()
```

Comment:
This will read the entire file into memory, which would be bad for the files that are being hashed.

In python 3.11, you can do [`hashlib.file_digest`](https://docs.python.org/3/library/hashlib.html#hashlib.file_digest):

```suggestion
        with open(file_name, "rb") as f:
            return hashlib.file_digest(f, "sha256").hexdigest()
```

But I think you are on 3.10 now so that'd look more like:

```suggestion
        # TODO: use hashlib.file_digest when we upgrade to python>=3.11
        hashobj = hashlib.sha256()
        buf = bytearray(2**18)
        view = memoryview(buf)

        with open(file_name, "rb") as f:
            while True:
                size = f.readinto(buf)
                if size == 0:
                    break  # EOF
                hashobj.update(view[:size])
        return hashobj.hexdigest()
```

I also don't know that this needs to be a method, and could be a utility.

Reply:
I would probably also go for a faster hash like md5 here
```suggestion
        # TODO: use hashlib.file_digest when we upgrade to python>=3.11
        hashobj = hashlib.md5()
        buf = bytearray(2**18)
        view = memoryview(buf)
        with open(file_name, "rb") as f:
            while True:
                size = f.readinto(buf)
                if size == 0:
                    break  # EOF
                hashobj.update(view[:size])
        return hashobj.hexdigest()
```

Reply:
why do you have 
```
        buf = bytearray(2**18)
        view = memoryview(buf)
```

Reply:
So that you reuse the buffer instead of allocating new ones.

Totally forgot to mention this is based off the definition of `hashlib.file_digest` in python 3.11